src/migrateDashboard.ts

```diff
@@ -34,12 +37,19 @@ import { _getLegacyWidgetPluginKey } from "./_getLegacyWidgetPluginKey";
  */
 export function migrateDashboard(
   legacyDashboardState: LegacyDashboardState,
-  servers: { [serverKey: string]: { dataModel: DataModel; url: string } },
-  keysOfWidgetPluginsToRemove?: string[],
-): DashboardState<"serialized"> {
+  {
+    servers,
+    keysOfWidgetPluginsToRemove,
+    doesReportIncludeStacks,
+  }: {
+    servers: { [serverKey: string]: { dataModel: DataModel; url: string } };
+    keysOfWidgetPluginsToRemove?: string[];
+    doesReportIncludeStacks?: boolean;
+  },
+): [DashboardState<"serialized">, DashboardErrorReport["pages"]?] {
   const pages: { [pageKey: string]: DashboardPageState<"serialized"> } = {};
   const body = legacyDashboardState.value.body;
-
+  const pagesHavingErrors: DashboardErrorReport["pages"] = {};
```

Comment:
If it is called `pagesSomething`, then the variable should be a `DashboardPage[]`. In this case, it is a map of errors. Why not avoiding this confusion by doing:

```suggestion
  const errorReport: Omit<DashboardErrorReport, "folderId" | "folderName"> = {};
```

This way, `migrateDashboard` would return a partial `DashboardErrorReport` instead of a `DashboardPagesErrorReport`, which I also find more logical. We could just add a comment in the signature explaining that `folderId` and `folderName` have to be added by the caller.